scimono-compliance-tests/src/main/java/com/sap/scimono/scim/system/tests/SCIMComplianceTest.java

```diff
@@ -70,11 +72,29 @@
 
   protected SCIMComplianceTest() {
     SCIMClientService scimClientService = configureScimClientService(SERVICE_URL);
-    userRequest = scimClientService.buildUserRequest();
-    groupRequest = scimClientService.buildGroupRequest();
-    schemaRequest = scimClientService.buildSchemaRequest();
+    SCIMRequest.Builder builder = SCIMRequest.newBuilder();
+    populateCustomHeaders(builder);
+
+    userRequest = scimClientService.buildUserRequest(builder);
+    groupRequest = scimClientService.buildGroupRequest(builder);
+    schemaRequest = scimClientService.buildSchemaRequest(builder);
   }
 
+  private void populateCustomHeaders(SCIMRequest.Builder newBuilder) {
```

Comment:
What about the extensions which also use `buildUserRequest`, `buildGroupRequest` and `buildSchemaRequest`?
If you change the method signature to:
```java
public static SCIMRequest.Builder populateCustomHeaders() {
 
}
```

You can easily pass the headers to them.